node/pkg/processor/processor.go

```diff
@@ -117,7 +148,7 @@ type Processor struct {
 	// Runtime state
 
 	// gs is the currently valid guardian set
-	gs *common.GuardianSet
+	gs atomic.Pointer[common.GuardianSet]
```

Comment:
could also just use `gst.Get()` everywhere instead of maintaining another variable.